### PR TITLE
Include USWDS assets and overrides

### PR DESCRIPTION
**Pull request by jessieay**
- Downloaded via NPM
- Included in subdir to mitigate any namespacing problems with assets

### CODE REVIEW

**Comment by jessieay** [comment]
Have no idea if this will work, just trying to replicate what we have in Micropurchase.

Here is my thought process so far:

Include all USWDS in subdir. So, scss src files in `app/assets/stylesheets/uswds/` and then overrides in `app/assets/stylesheets/uswds_overrides`

Each of these has a manifest called `all.scss` so in install instructions we can tell people to include `uswds/all` above `uswds_overrides/all` similar to https://github.com/18F/micropurchase/blob/develop/app/assets/stylesheets/application.scss#L2

Questions:
- does this dir hierarchy make sense?
- do we have an app that uses uswds js? MP doesn't. The design standards [README](https://github.com/18F/web-design-standards) says 

>  require('uswds') will load all of the Draft U.S. Web Design Standard's JavaScript onto the page. 

Not sure how we should deal with this in rails...
- 8k+ line PR..what could possibly go wrong?

**Comment by jessieay** [comment]
Also wondering if we should include in `vendor` dir in gem (rather than `app`) since that is where I'd put these in a Rails app. 

I like the simplicity of how this gem wraps some assets for rails asset pipeline: https://github.com/derekprior/momentjs-rails (it has been dowloaded over a million times: https://rubygems.org/gems/momentjs-rails/versions/2.11.1 )

**Comment by pkarman** [comment]
I like the idea of uswds stuff in `vendor/` -- that seems consistent with how Rails does stuff in general.

See my comments on #2 for other structure ideas.

I don't have an existing app that uses uswds js; that's what we should build in our tests, a little mini app that uses that for a 'hello world' type thing.

File: app/assets/config/uswds_rails_manifest.js [hunk 1]
```diff
@@ -1,2 +0,0 @@
-//= link_directory ../javascripts/uswds/rails .js
```
Comment (jessieay) [inline app/assets/config/uswds_rails_manifest.js:1]: do we need this file? Seems unusual to have a `config` file in `app/assets` ?

Comment (pkarman) [inline app/assets/config/uswds_rails_manifest.js:1]: agreed, it does seem odd. It was generated by Rails as boilerplate. If tests pass w/o it, I'm happy to see it deleted.

**Comment by pkarman** [comment]
our hello world test is failing at travis. Time to change that test?

can we fetch the main uswds css file(s) instead? Just a simple 200 response for the right file seems ok to me.

Once tests are passing, I'm +1 to just merge this as-is and then iterate in smaller chunks as we increase tests.

**Pull request merged**